src/linux.rs

```diff
@@ -60,24 +60,59 @@ impl Output {
 
 #[cfg(target_os = "linux")]
 pub(crate) fn get_disk_id() -> Result<String, HWIDError> {
-    let mut com = Command::new("sh");
-    com.arg("-c").arg("lsblk -f -J -o NAME,MOUNTPOINT,UUID");
+    let output = run_command("lsblk -f -J -o NAME,MOUNTPOINT,UUID")?;
 
-    let output = com.output()?;
-
-    let output_string = String::from_utf8(output.stdout)?;
+    let output_string = String::from_utf8(output.into())?;
     let parsed: Output = serde_json::from_str(output_string.as_str())?;
     let uuid = parsed.get_root()?;
     Ok(uuid)
 }
 
+#[cfg(target_os = "linux")]
+fn run_command(command: &str) -> Result<String, HWIDError> {
+    let mut cmd = Command::new("sh");
+    let cmd = cmd.arg("-c")
+        .arg(command);
+
+    let output = cmd.output()?;
+    if !cmd.status()?.success() {
+        return Err(HWIDError::new(
+            &format!("Failed to run command: {command}"),
+            &String::from_utf8(output.stderr.into())?,
+        ));
+    }
+
+    Ok(String::from_utf8(cmd.output()?.stdout)?)
+}
+
+#[cfg(target_os = "linux")]
+fn get_mac_addressof_interface(interface_name: &str) -> Result<String, HWIDError> {
+    run_command(&format!("cat /sys/class/net/{interface_name}/address"))
+}
+
 #[cfg(target_os = "linux")]
 pub(crate) fn get_mac_address() -> Result<String, HWIDError> {
-    let mut com = Command::new("sh");
-    com.arg("-c")
-        .arg("cat /sys/class/net/$(ip route show default | awk '/default/ {print $5}')/address");
-    let output = com.output()?;
-    Ok(String::from_utf8(output.stdout)?)
+    // First check for first cable internet interface connected on hardware,
+    // if the ethernet cable is not available we find the first wifi interface.
+    let first_cable_interface_names = [vec!["eth0".to_string()]
+        , (0..10)
+            .map(|x| format!("enp{x}s0"))
+            .collect::<Vec<String>>()
+        , vec!["wlan0".to_string()]
+        , (0..10)
+            .map(|x| format!("wlp{x}s0"))
+            .collect::<Vec<String>>()
+    ].concat();
+
+    for interface_name in first_cable_interface_names {
+        let result = get_mac_addressof_interface(&interface_name);
+        if result.is_ok() {
+            return result;
+        }
+    }
+
+    // If everything just fails, we get the default network interface
+    get_mac_addressof_interface(&run_command("ip route show default | awk '/default/ {print $5}'")?)
```

Comment:
Note that this command can produce a multi-line output, we probably want to split it to get the first matched `<interface>[\n]`.

Reply:
> Note that this command can produce a multi-line output, we probably want to split it to get the first matched `<interface>[\n]`.

That was the original code before this PR, unrelated to this change.

Reply:
Okay, here's the issue for that: https://github.com/Taptiive/machineid-rs/issues/16